app/jinja_filters.py

```diff
@@ -78,13 +78,13 @@ def format_unit(
     # mass-metric-ton no longer supported for en_GB and related locales, but still present in business schema and allowed in validator,
     # until removed from schema we substitute mass-tonne for mass-metric-ton before format unit
     measurement_unit = "mass-tonne" if unit == "mass-metric-ton" else unit
-
+    formatted_value = value if value in [1, 2, ""] else 0
```

Comment:
instead of using `format_unit` like this to get the unit name with a dummy value and then replace the value, I think it would be better to use the babel function that `format_unit` calls to get the unit name

Then you could just do
```
unit_name = units.get_unit_name(
    measurement_unit=measurement_unit,
    length=length,
    locale=flask_babel.get_locale(),
)
return f"{value}{unit_name}"
```

Reply:
In validator we verify that the maximum number of decimal places allowed is 6, so instead of the above, I think this could be another valid way to achieve the same thing
```
formatted_unit = units.format_unit(
   value=value,
   measurement_unit=measurement_unit,
   length=length,
   locale=flask_babel.get_locale(),
   format="#.######"
)
```
possibly cleaner but I guess its still just another workaround for them not offering the quantization option that the other format methods do

Reply:
When you provide a format, is locale not ignored?

Reply:
Also, why do we do `formatted_value = value if value in [1, 2, ""] else 0`, that is confusing.

Reply:
So something similar to this would be valid?:
```python
    decimal_part = None

    if value:
        split_value = str(value).split(".")
        value = int(split_value[0])
        decimal_part = split_value[1] if len(split_value) > 1 else None

    formatted_unit: str = units.format_unit(
        value=value,
        measurement_unit=measurement_unit,
        length=length,
        locale=flask_babel.get_locale(),
    )
    if decimal_part:
        formatted_unit_list = formatted_unit.split(" ")
        formatted_unit_list[0] = f"{formatted_unit_list[0]}.{decimal_part}"
        return " ".join(formatted_unit_list)
    return formatted_unit
```
but then, as Mick mentioned, if it was 1001.1111, it would become `1,001.1111 kilometre` not `1,001.1111 kilometres`..

Reply:
This might fail with a length of `narrow` where the format would be `1,001km`

There might be a way to do this with without the split, looking at the library, if you used one of the babel functions to get the precision of the number, you could split the pattern on "." and add that number of 0s to it

e.g. 
`numbers.format_decimal("1230.2400", locale="en", format="#,###.0000")` preserves the trailing zeros

Reply:
Hacky, but former gurus of this time already introduced similar hacky code before:
```python
    if unit_length == "long":
        unit_label = format_unit(
            value=2, unit=unit, length=unit_length, locale=locale
        ).replace("2 ", "")
```

Reply:
That is okay, and not brittle, when we just mutate the string, like we are doing, it becomes brittle and prone to error, I think we need to try do it via providing our own format but only modifying the decimal bit. Don't think there is a perfect solution, we just need to go with least prone to error.

Reply:
> Ah true, a dynamic replacement would be better, the workaround here might be viable to some extent [python-babel/babel#90](https://github.com/python-babel/babel/issues/90) with something like `pattern = pattern.replace(".#", ".######")` but this isn't going to fix trailing zeroes, I can't immediately see a way to do that with the babel library other than formatting the integer part and manually adding the decimal part

This is our best bet I think. We just need to them adjust to fix trailing zeros.